modules/nextflow/src/main/groovy/nextflow/dag/DagRenderer.groovy

```diff
@@ -24,10 +24,19 @@ import java.nio.file.Path
  * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
  * @author Mike Smoot <mes@aescon.com>
  */
-interface DagRenderer {
+trait DagRenderer {
 
```

Comment:
This can still be an interface and provide the methods below as `default` methods https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html